packages/discovery-react-components/src/utils/document/__tests__/processDoc.spec.tsx

```diff
@@ -31,6 +33,19 @@ describe('processDoc', () => {
     doc = await processDoc(contractData.results[0], { sections: true });
   });
 
+  it('does not mutate the passed document data and metadata property are not be present in it', () => {
+    expect(doc.metadata).toHaveLength(4);
+    expect(get(contractData.results[0], 'enriched_html[0].contract.metadata')).toBeUndefined();
```

Comment:
I would prefer setting up a test that did a deep clone of the original data, then doing deep compare of original data against clone. That way we can ensure that _nothing_ is changed, not just `metadata`.

Reply:
That will just check if `lodash clonedeep` did  the job or not,  `processDoc` function already has a clonedeep which clones  the original data, make some transformations to it and return  `data` which has only 4 props.  
```
{styles: "/**/↵ .Arial_BoldMT_Black_ }↵/**/"
metadata: (2) [{…}, {…}]
sections: (237) [{…}, {…}, …]
itemMap: {byItem: {…}, bySection: {…}}

So comparing original data and returned data would always have a difference so I think it is necessary to check explicitly if original data does not have `metadata` / `attributes`, `relations` after calling `processDoc` function (which already has a deepcloned original data). Thoughts?

Reply:
Right, but wasn't the original issue that we **weren't** doing a deep clone in `processDoc` and instead updating the incoming document data object?

Reply:
Right! We weren't doing a cloneDeep earlier, So now that `processDoc` has it, input data can be checked after running a `processDoc` function, and then check explicitly if `metadata` field is not present in the original input data.

Reply:
What I don't like is that this test is too specific -- it only checks if we are adding `metadata`. But if future changes pollute the incoming document data with some other property? I want a general test that would catch such a bug.

Reply:
Done! Added another check for entire data. Also kept metadata check since this specific issue is fixed in this PR.